### ISSUE TITLE
Can't install Public and Private Ingress

### ISSUE THREAD

**Issue by bradenwright** [solo-io/gloo#3578]
**Describe the bug**
I'd like to have 2 different endpoints, typically to do this I'd install a helm chart 2 times with different values.:

```
gateway:
  enabled: false
ingress:
  enabled: true
  requireIngressClass: true
  customIngressClass: gloo-public
```
```
gateway:
  enabled: false
ingress:
  enabled: true
  requireIngressClass: true
  customIngressClass: gloo-private
ingressProxy:
  service:
    extraAnnotations:
      cloud.google.com/load-balancer-type: "Internal"
```

I get an error b/c resources like the [service name of ingressproxy are hard-coded in the helm chart](https://github.com/solo-io/gloo/blob/master/install/helm/gloo/templates/13-ingress-proxy-service.yaml#L8) 

Typically I use prepend `{{ template "gloo.fullname" . }}` to resources to avoid the conflict.  I just wanted to make sure there was a more gloo native approach way to accomplish this.  If an approach is agreed upon I don't mind pushing a PR.

**To Reproduce**
1. helm upgrade --install gloo-public gloo/gloo -f values-gloo-public.yaml
2. helm upgrade --install gloo-private gloo/gloo -f values-gloo-private.yaml

**Expected behavior**

2 different installs of Gloo 1 with a public Service LB, and the other with a private.

**Additional context**
Add any other context about the problem here, e.g.
- Gloo version
- Kubernetes version
- Helm chart

**Comment by bradenwright** [comment]
Any update on this, it will just be a good amount of work to go thru the chart and make it so it can be installed multiple times so I don't want to do that unless (1) there isn't already a way (2) the approach / PR is gonna be accepted.

**Comment by pbchekin** [comment]
@bradenwright, you need to install Gloo Ingress instances to different namespaces (by default, the namespace is `gloo`). However, even installed to different namespaces, they will not together because of #3587.

**Comment by bradenwright** [comment]
@pbchekin thanks for opening the other issue / looking into it!  While I'd rather install into the same namespace that's not a show stopper for me.  So when #3587 is fixed I'll start using Gloo, b/c I do need public / private in the same cluster.

**Comment by kdorosh** [comment]
@bradenwright  fyi https://github.com/solo-io/gloo/issues/3587 has been fixed and released in the most recent 1.5 beta release
